### PR TITLE
Added RegSeg to codebase

### PR DESCRIPTION
**Pull request by NatanBagrov**

### CODE REVIEW

File: src/super_gradients/recipes/cityscapes_regseg48.yaml [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    - PixelAccuracy:
+        ignore_label: ${dataset_params.cityscapes_ignored_label}
+    - IoU:
+        num_classes: 20
+        ignore_index: ${dataset_params.cityscapes_ignored_label}
+
+  metric_to_watch: IoU
+  greater_metric_to_watch_is_better: True
+
+  sg_logger: "wandb_sg_logger"
```
Comment (shaydeci) [inline src/super_gradients/recipes/cityscapes_regseg48.yaml:122]: Use default sg_logger as it will crash for our users (just remove this part).

**Review by shaydeci — CHANGES_REQUESTED** [review]
2 minor comments (inline).

File: src/super_gradients/recipes/cityscapes_regseg48.yaml (hunk 1 above)
Comment (NatanBagrov) [reply src/super_gradients/recipes/cityscapes_regseg48.yaml]: Done here [7ca5929](https://github.com/Deci-AI/super-gradients/pull/125/commits/7ca5929b79c96d68d8336de291503829a11b7f44)

File: src/super_gradients/recipes/cityscapes_regseg48.yaml [hunk 2]
```diff
@@ -0,0 +1,131 @@
+#  RegSeg segmentation training example with Cityscapes dataset.
+#  Reproduction of paper: Rethink Dilated Convolution for Real-time Semantic Segmentation.
+#
+#  Usage RegSeg48:
+#      python -m torch.distributed.launch --nproc_per_node=4 src/super_gradients/examples/train_from_recipe_example/train_from_recipe.py --config-name=regseg48_cityscapes
+#
+#
+#  Validation mIoU - Cityscapes, training time:
```
[the remark this answers is not in the document]
Comment (NatanBagrov) [reply src/super_gradients/recipes/cityscapes_regseg48.yaml:8]: Dene here [6f32d60](https://github.com/Deci-AI/super-gradients/pull/125/commits/6f32d60b400e49ce95c454e7e4c224ae09c347fd)

**Review by shaydeci — APPROVED** [review]
LGTM

**Pull request merged**